src/nvme/linux.c

```diff
@@ -212,9 +214,39 @@ static int nvme_get_telemetry_log(int fd, bool create, bool ctrl, bool rae,
 		return 0;
 	}
 
-	/* dalb3 >= dalb2 >= dalb1 */
-	size = (le16_to_cpu(telem->dalb3) + 1) * xfer;
-	tmp = realloc(log, size);
+	switch (da) {
+	case 1:
+	case 2:
+	case 3:
+		/* dalb3 >= dalb2 >= dalb1 */
+		*size = (le16_to_cpu(telem->dalb3) + 1) * xfer;
+		break;
+	case 4:
+		err = nvme_identify_ctrl(fd, &id_ctrl);
+		if (err) {
+			perror("identify-ctrl");
+			errno = EINVAL;
+			goto free;
+		}
+
+		if ((id_ctrl.lpa & 0x40)) {
+			*size = (le32_to_cpu(telem->dalb4) + 1) * xfer;
+		}
+		else {
```

Comment:
Please move the else to the closing braket `} else {`

Reply:
Fixed with latest commit.